torch/distributions/negative_binomial.py

```diff
@@ -0,0 +1,83 @@
+from numbers import Number
+import torch
+import torch.nn.functional as F
+from torch.distributions import constraints
+from torch.distributions.distribution import Distribution
+from torch.distributions.utils import broadcast_all, probs_to_logits, lazy_property, logits_to_probs
+
+
+class NegativeBinomial(Distribution):
+    r"""
+        Creates a Negative Binomial distribution parameterized by
+        a Bernoulli trial with probability `probs` of success that
+        occur until we observe `total_count` number of failures.
+
+    Args:
+        total_count (int or Tensor): number of negative Bernoulli trials to stop
+        probs (Tensor): Event probabilities of success
+        logits (Tensor): Event log-odds for probabilities of success
+    """
+
+    arg_constraints = {'total_count': constraints.nonnegative_integer,
+                       'probs': constraints.unit_interval}
+    support = constraints.nonnegative_integer
+
+    def __init__(self, total_count=1, probs=None, logits=None, validate_args=None):
+        if (probs is None) == (logits is None):
+            raise ValueError("Either `probs` or `logits` must be specified, but not both.")
+        if probs is not None:
+            self.total_count, self.probs, = broadcast_all(total_count, probs)
+            self.total_count = self.total_count.type_as(self.logits)
+            is_scalar = isinstance(self.probs, Number)
+        else:
+            self.total_count, self.logits, = broadcast_all(total_count, logits)
+            self.total_count = self.total_count.type_as(self.logits)
+            is_scalar = isinstance(self.logits, Number)
```

Comment:
Hmm I believe this will always be false since `broadcast_all` only returns tensors.